### PR TITLE
Create a security policy

### PR DESCRIPTION
**Pull request by martinthomson**
A simple document describing how we handle advisories is probably long overdue.

### CODE REVIEW

File: SECURITY.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+## Supported Versions
+
+Support for neqo is based on the Firefox version in which it has landed.
+Versions of neqo in [current versions of Firefox](https://whattrainisitnow.com/calendar/) are actively supported.
+
+The version of neqo that is active can be found in the Firefox repositories:
+[release](https://hg.mozilla.org/mozilla-unified/file/release/third_party/rust/neqo-transport/Cargo.toml),
+[beta](https://hg.mozilla.org/mozilla-unified/file/beta/third_party/rust/neqo-transport/Cargo.toml),
+[trunk/central](https://hg.mozilla.org/mozilla-unified/file/central/third_party/rust/neqo-transport/Cargo.toml),
+and [ESR 115](https://hg.mozilla.org/mozilla-unified/file/esr115/third_party/rust/neqo-transport/Cargo.toml).
```
Comment (larseggert) [inline SECURITY.md:14]:
```suggestion

- [release](https://hg.mozilla.org/mozilla-unified/file/release/third_party/rust/neqo-transport/Cargo.toml),
- [beta](https://hg.mozilla.org/mozilla-unified/file/beta/third_party/rust/neqo-transport/Cargo.toml),
- [trunk/central](https://hg.mozilla.org/mozilla-unified/file/central/third_party/rust/neqo-transport/Cargo.toml),
- and [ESR 115](https://hg.mozilla.org/mozilla-unified/file/esr115/third_party/rust/neqo-transport/Cargo.toml).
```

File: SECURITY.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+The listed version in these files corresponds to [tags](https://github.com/mozilla/neqo/tags) on this repository.
+Releases do not always correspond to a branch.
+
+We welcome reports of security vulnerabilities in any of these released versions or the latest code on the `main` branch.
+
+## Reporting a Vulnerability
+
+To report a security problem with create a bug in Mozilla's Bugzilla instance in the [Core :: Networking](https://bugzilla.mozilla.org/enter_bug.cgi?product=Core&component=Networking) component.
```
Comment (larseggert) [inline SECURITY.md:22]:
```suggestion
To report a security problem with neqo, create a bug in Mozilla's Bugzilla instance in the [Core :: Networking](https://bugzilla.mozilla.org/enter_bug.cgi?product=Core&component=Networking) component.
```

File: SECURITY.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+IMPORTANT: For security issues, please make sure that you check the box labelled "Many users could be harmed by this security problem".
+We advise that you check this option for anything that involves anything security-relevant, including memory safety, crashes, race conditions, and handling of confidential information.
```
Comment (larseggert) [inline SECURITY.md:25]:
```suggestion
**IMPORTANT: For security issues, please make sure that you check the box labelled "Many users could be harmed by this security problem".**
We advise that you check this option for anything that involves anything security-relevant, including memory safety, crashes, race conditions, and handling of confidential information.
```

File: SECURITY.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+Review Mozilla's [guides on bug reporting](https://bugzilla.mozilla.org/page.cgi?id=bug-writing.html) before you open a bug.
+
+Mozilla operates a [bug bounty program](https://www.mozilla.org/en-US/security/bug-bounty/), for which this project is elligible.
```
Comment (larseggert) [inline SECURITY.md:29]:
```suggestion
Mozilla operates a [bug bounty program](https://www.mozilla.org/en-US/security/bug-bounty/), for which this project is eligible.
```

**Review by larseggert — DISMISSED** [review]
LGTM modulo some nits

File: SECURITY.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+Support for neqo is based on the Firefox version in which it has landed.
+Versions of neqo in [current versions of Firefox](https://whattrainisitnow.com/calendar/) are actively supported.
+
+The version of neqo that is active can be found in the Firefox repositories:
+
+- [release](https://hg.mozilla.org/mozilla-unified/file/release/third_party/rust/neqo-transport/Cargo.toml),
+- [beta](https://hg.mozilla.org/mozilla-unified/file/beta/third_party/rust/neqo-transport/Cargo.toml), and
+- [trunk/central](https://hg.mozilla.org/mozilla-unified/file/central/third_party/rust/neqo-transport/Cargo.toml),
+- [ESR 115](https://hg.mozilla.org/mozilla-unified/file/esr115/third_party/rust/neqo-transport/Cargo.toml).
+The listed version in these files corresponds to [tags](https://github.com/mozilla/neqo/tags) on this repository.
```
Comment (martinthomson) [inline SECURITY.md:16]:
```suggestion

The listed version in these files corresponds to [tags](https://github.com/mozilla/neqo/tags) on this repository.
```

**Review by larseggert — APPROVED** [review]

**Pull request merged**